modules/core/include/opencv2/core/softfloat.hpp

```diff
@@ -0,0 +1,240 @@
+    static softfloat  inf() { return softfloat::fromRaw( 0xFF << 23 ); }
+    static softfloat  nan() { return softfloat::fromRaw( 0x7fffffff ); }
+    static softfloat  one() { return softfloat::fromRaw(  127 << 23 ); }
+
+    uint32_t v;
+};
+
+/*----------------------------------------------------------------------------
+*----------------------------------------------------------------------------*/
+
+struct CV_EXPORTS softdouble
+{
+public:
+    softdouble() { }
+    softdouble( const softdouble& c) { v = c.v; }
+    softdouble& operator=( const softdouble& c )
+    {
+        if(&c != this) v = c.v;
+        return *this;
+    }
+    static softdouble fromRaw( const uint64_t a ) { softdouble x; x.v = a; return x; }
+
+    explicit softdouble( const uint32_t );
+    explicit softdouble( const uint64_t );
+    explicit softdouble( const  int32_t );
+    explicit softdouble( const  int64_t );
+    explicit softdouble( const double a ) { Cv64suf s; s.f = a; v = s.u; }
+
+    operator softfloat() const;
+    operator double() const { Cv64suf s; s.u = v; return s.f; }
+
+    softdouble operator + (const softdouble&) const;
+    softdouble operator - (const softdouble&) const;
+    softdouble operator * (const softdouble&) const;
+    softdouble operator / (const softdouble&) const;
+    softdouble operator % (const softdouble&) const;
+    softdouble operator - () const { softdouble x; x.v = v ^ (1ULL << 63); return x; }
+
+    softdouble& operator += (const softdouble& a) { *this = *this + a; return *this; }
+    softdouble& operator -= (const softdouble& a) { *this = *this - a; return *this; }
+    softdouble& operator *= (const softdouble& a) { *this = *this * a; return *this; }
+    softdouble& operator /= (const softdouble& a) { *this = *this / a; return *this; }
+    softdouble& operator %= (const softdouble& a) { *this = *this % a; return *this; }
+
+    bool operator == ( const softdouble& ) const;
+    bool operator != ( const softdouble& ) const;
+    bool operator >  ( const softdouble& ) const;
+    bool operator >= ( const softdouble& ) const;
+    bool operator <  ( const softdouble& ) const;
+    bool operator <= ( const softdouble& ) const;
+
+    bool isNaN() const { return (v & 0x7fffffffffffffff)  > 0x7ff0000000000000; }
+    bool isInf() const { return (v & 0x7fffffffffffffff) == 0x7ff0000000000000; }
+
+    static softdouble zero() { return softdouble::fromRaw( 0 ); }
+    static softdouble  inf() { return softdouble::fromRaw( (uint_fast64_t)(0x7FF) << 52 ); }
+    static softdouble  nan() { return softdouble::fromRaw( CV_BIG_INT(0x7FFFFFFFFFFFFFFF) ); }
+    static softdouble  one() { return softdouble::fromRaw( (uint_fast64_t)( 1023) << 52 ); }
+
+    uint64_t v;
+};
+
+/*----------------------------------------------------------------------------
+*----------------------------------------------------------------------------*/
+
+CV_EXPORTS int_fast64_t round( const softfloat&  a );
```

Comment:
round and trunc should return int, not int_fast64_t. instead of trunc(), implement "operator int() const" inside the classes. floor() and ceil() would be also useful